src/server/pfs/db/driver.go

```diff
@@ -1331,7 +1331,12 @@ func (r *fileReader) Read(data []byte) (int, error) {
 			break
 		}
 		client := client.APIClient{BlockAPIClient: r.blockClient}
-		r.reader, err = client.GetBlock(blockRef.Hash, uint64(r.offset), uint64(r.size))
+		sizeLeft := r.size
+		// e.g. sometimes a reader is constructed of size 0
+		if sizeLeft != 0 {
+			sizeLeft -= r.sizeRead
+		}
+		r.reader, err = client.GetBlock(blockRef.Hash, uint64(r.offset), uint64(sizeLeft))
```

Comment:
Oh, I branched off that branch (before it landed) because I needed the TF example working before I could make a test for it. I need to merge in master on this branch, and then the diff will reflect the right things.